[BUG] Unable to create custom PromptTemplate

**Describe the bug**
I am trying langchain4J in Android and Upon creation new PromptTemplate as below, I see exception java.lang.ExceptionInInitializerError

```
 PromptTemplate prompt = new PromptTemplate(
                "Read and understand the conversation between the User and the AI. " +
                        "Then, analyze the new query from the User. " +
                        "Identify all relevant details, terms, and context from both the conversation and the new query. " +
                        "Reformulate this query into a clear, concise, and self-contained format suitable for information retrieval.\n" +
                        "\n" +
                        "Conversation:\n" +
                        "{{chatMemory}}\n" +
                        "\n" +
                        "User query: {{query}}\n" +
                        "\n" +
                        "It is very important that you provide only reformulated query and nothing else! " +
                        "Do not prepend a query with anything!"
        );
```
        


**Log and Stack trace**
java.lang.ExceptionInInitializerError
                                                                                                    	at dev.langchain4j.model.input.DefaultPromptTemplateFactory.create(DefaultPromptTemplateFactory.java:18)
                                                                                                    	at dev.langchain4j.model.input.DefaultPromptTemplateFactory.create(DefaultPromptTemplateFactory.java:14)
                                                                                                    	at dev.langchain4j.model.input.PromptTemplate.<init>(PromptTemplate.java:62)
                                                                                                    	at dev.langchain4j.model.input.PromptTemplate.<init>(PromptTemplate.java:51)


 Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 10
                                                                                                    \{\{(.+?)}}
                                                                                                              ^
**To Reproduce**
Just try to create the PromptTemplate 

**Expected behavior**
PromptTemplate  should be created.

**Please complete the following information:**
- LangChain4j version: 0.30
- LLM(s) used: Azure openAI
- Java version: Android Q

> The `java.lang.ExceptionInInitializerError` you're encountering seems to be related to the static initialization phase of the `PromptTemplate` class in LangChain4j, specifically when it tries to initialize a `PromptTemplateFactory`. This error can occur due to several reasons, such as service loading failure, exceptions in factory implementations, or compatibility issues with Android.
> 
> To address this issue, you should:
> 
> 1. **Verify Service Provider Configuration**: Ensure any custom `PromptTemplateFactory` implementations are correctly specified in `META-INF/services`. The service provider configuration file should be named `dev.langchain4j.spi.prompt.PromptTemplateFactory` and contain the fully qualified name of your implementation class.
> 2. **Review Factory Implementations**: Make sure that any custom implementations of `PromptTemplateFactory` do not throw exceptions during instantiation and are compatible with Android.
> 3. **Check Android Compatibility**: Ensure all Java features used in your `PromptTemplateFactory` implementations and the `loadFactories` method are compatible with the Android version you are targeting.
> 4. **Debug Static Initialization**: Consider adding logging or try-catch blocks around the static initialization code to identify the exact cause of the error.
> 
> If these steps do not resolve the issue, you might need to provide a custom initialization path for Android that avoids problematic static initialization or uses alternatives compatible with Android.
> 
> #### Details
> _To continue the conversation, mention @dosu._

The root cause seems to be due to 

 private static final Pattern VARIABLE_PATTERN = Pattern.compile("\\{\\{(.+?)}}");
 
Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 10 {{(.+?)}} ^ this is happening can u see why

--- a/langchain4j-core/src/main/java/dev/langchain4j/model/input/DefaultPromptTemplateFactory.java
+++ b/langchain4j-core/src/main/java/dev/langchain4j/model/input/DefaultPromptTemplateFactory.java
@@ -20,7 +20,8 @@ class DefaultPromptTemplateFactory implements PromptTemplateFactory {
 
     static class DefaultTemplate implements Template {
 
-        private static final Pattern VARIABLE_PATTERN = Pattern.compile("\\{\\{(.+?)}}");
+        @SuppressWarnings("RegExpRedundantEscape")
+        private static final Pattern VARIABLE_PATTERN = Pattern.compile("\\{\\{(.+?)\\}\\}");
 
         private final String template;
         private final Set<String> allVariables;

